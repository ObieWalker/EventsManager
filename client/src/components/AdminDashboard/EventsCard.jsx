import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Col } from 'react-materialize';

/**
 *
 *
 * @class EventsCard
 * @extends {Component}
 */
class EventsCard extends Component {
  /**
   * Creates an instance of EventsCard.
   * @param {any} props
   * @memberof EventsCard
   */
  constructor(props) {
    super(props);

    this.handleCancel = this.handleCancel.bind(this);
  }
  /**
 * @returns {*} null
 *
 * @param {any} id
 * @memberof EventsCard
 */
  handleCancel(id) {
    this.props.cancelEvent(id);
  }
  /**
 *
 *
 * @returns {object} event
 * @memberof EventsCard
 */
  render() {
    const { event } = this.props;
    return (
      <div style={{ margin: '5%' }}>
        <Col s={10} m={4}>
          <div className="card elegant-color white-text"
            style={{
              width: '18rem', padding: '5%', borderRadius: '10%', height: '200px'
            }}>
            <div className="card-body">
              <h5 className="card-title white-text">{event.eventType}</h5>
              <h6 className="card-subtitle mb-2 text-muted white-text">
                @ {event.Center.name}</h6>
              <p className="card-text white-text">{event.date}</p>
              <p className="card-text white-text">
              Estimated guest No.: {event.guestNo}</p>
            </div>
            {!event.isCancelled ?
              <button className="btn btn-danger btn-sm btn-rounded"
                onClick={this.handleCancel.bind(this, event.id)}
              >Cancel</button> :
              <p className="red-text">
              This event has been canceled by an admin.</p>
            }
          </div>
        </Col>
      </div>
    );
  }
}

EventsCard.propTypes = {
  event: PropTypes.object,
  cancelEvent: PropTypes.func,
  center: PropTypes.string,
  eventType: PropTypes.string,
  date: PropTypes.instanceOf(Date),
  email: PropTypes.string,
  getAllCenters: PropTypes.func,
  guestNo: PropTypes.number
};

export default EventsCard;
