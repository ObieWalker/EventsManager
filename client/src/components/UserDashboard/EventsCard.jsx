import React from 'react';
import PropTypes from 'prop-types';
import { Col } from 'react-materialize';

const EventCard = (props) => {
  const { event } = props;
  const cutOffTime = Date.now() + 86400000;
  const eventDate = new Date(event.date);
  const today = Date.now();
  if (event) {
    return (
      <div style={{ margin: '1%', float: 'left', display: 'inlineBlock' }}>
        <Col s={10} m={4}>
          <div className="col 12">
            <div
              className="card elegant-color white-text"
              style={{
                width: '18rem',
                padding: '5%',
                borderRadius: '10%',
                height: '260px'
              }}
            >
              <div className="card-body">
                <div style={{ minHeight: '9rem' }}>
                  <h5 className="card-title white-text">{event.eventType}</h5>

                  <h6 className="card-subtitle mb-2 text-muted white-text">
                    {event.Center
                      ? `Center: ${event.Center.name}`
                      : 'This center has been removed'}
                  </h6>
                  <p className="card-text white-text">Date: {event.date}</p>
                  <p className="card-text white-text">
                    {event.Center && `Address: ${event.Center.address}`}
                  </p>
                  <p className="card-text white-text">
                    Guest Estimate: {event.guestNo}
                  </p>
                </div>
                <hr className="hr-light" />
                {event.isCancelled || !event.Center ? (
                  <h6 className="red-text">
                    Apologies!!! This event has been cancelled.
                  </h6>
                ) : (
                  <div>
                    {eventDate > today && (
                      <div>
                        {eventDate > cutOffTime ? (
                          <div>
                            <button
                              className="btn blue-gradient active btn-sm"
                              onClick={() => props.handleShowEditModal(event)}
                            >
                              Edit
                            </button>
                            <button
                              className="btn btn-primary btn-danger btn-sm"
                              onClick={() => props.handleDeleteEvent()}
                            >
                              Cancel{' '}
                            </button>
                          </div>
                        ) : (
                          <div>
                            <h6 className="red-text">
                              Unable to Cancel or Modify 24 hrs to event.
                            </h6>
                          </div>
                        )}
                      </div>
                    )}
                  </div>
                )}
              </div>
            </div>
          </div>
        </Col>
      </div>
    );
  }
};

EventCard.propTypes = {
  handleShowEditModal: PropTypes.func,
  handleDeleteEvent: PropTypes.func,
  event: PropTypes.object
};

export default EventCard;
