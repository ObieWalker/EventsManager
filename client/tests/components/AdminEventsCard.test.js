import React from 'react';
import expect from 'expect';
import { shallow } from 'enzyme';
import EventsCard from '../../src/components/AdminDashboard/EventsCard';

let mountedForm;
let props;
let wrapper;
const event = {
  Center: {}
};
const cancelEvent = jest.fn();
const handleCancel = jest.fn();
const center = '';
const eventType = '';
const date = '';
const email = '';
const getAllCenters = jest.fn();
let guestNo;

const getComponent = () => {
  if (!mountedForm) {
    props = {
      event,
      cancelEvent,
      handleCancel,
      center,
      eventType,
      date,
      email,
      getAllCenters,
      guestNo
    };
    mountedForm = shallow(<EventsCard {...props} />);
  }
  return mountedForm;
};

describe('Center Card Component', () => {
  beforeEach(() => {});

  it('renders component successfully', () => {
    wrapper = getComponent();
    expect(wrapper).toMatchSnapshot();
  });
  it('renders component successfully', () => {
    wrapper = getComponent();
    expect(wrapper.find('Col').length).toBe(1);
    expect(wrapper.find('button').length).toBe(1);
    expect(wrapper.find('div').length).toBe(3);
    expect(wrapper.find('p').length).toBe(2);
    expect(wrapper.find('h6').length).toBe(1);
    expect(wrapper.find('h5').length).toBe(1);
  });
});
