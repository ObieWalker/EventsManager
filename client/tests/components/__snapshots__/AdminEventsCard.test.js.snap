// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Center Card Component renders component successfully 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EventsCard
    cancelEvent={[MockFunction]}
    center=""
    date=""
    email=""
    event={
      Object {
        "Center": Object {},
      }
    }
    eventType=""
    getAllCenters={[MockFunction]}
    handleCancel={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Col
        m={4}
        s={10}
      >
        <div
          className="card elegant-color white-text"
          style={
            Object {
              "borderRadius": "10%",
              "height": "200px",
              "padding": "5%",
              "width": "18rem",
            }
          }
        >
          <div
            className="card-body"
          >
            <h5
              className="card-title white-text"
            />
            <h6
              className="card-subtitle mb-2 text-muted white-text"
            >
              @ undefined
            </h6>
            <p
              className="card-text white-text"
            />
            <p
              className="card-text white-text"
            >
              Estimated guest No.: 
            </p>
          </div>
          <button
            className="btn btn-danger btn-sm btn-rounded"
            onClick={[Function]}
          >
            Cancel
          </button>
        </div>
      </Col>,
      "style": Object {
        "display": "inlineBlock",
        "float": "left",
        "margin": "1%",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <div
          className="card elegant-color white-text"
          style={
            Object {
              "borderRadius": "10%",
              "height": "200px",
              "padding": "5%",
              "width": "18rem",
            }
          }
        >
          <div
            className="card-body"
          >
            <h5
              className="card-title white-text"
            />
            <h6
              className="card-subtitle mb-2 text-muted white-text"
            >
              @ undefined
            </h6>
            <p
              className="card-text white-text"
            />
            <p
              className="card-text white-text"
            >
              Estimated guest No.: 
            </p>
          </div>
          <button
            className="btn btn-danger btn-sm btn-rounded"
            onClick={[Function]}
          >
            Cancel
          </button>
        </div>,
        "m": 4,
        "s": 10,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="card-body"
            >
              <h5
                className="card-title white-text"
              />
              <h6
                className="card-subtitle mb-2 text-muted white-text"
              >
                @ undefined
              </h6>
              <p
                className="card-text white-text"
              />
              <p
                className="card-text white-text"
              >
                Estimated guest No.: 
              </p>
            </div>,
            <button
              className="btn btn-danger btn-sm btn-rounded"
              onClick={[Function]}
            >
              Cancel
            </button>,
          ],
          "className": "card elegant-color white-text",
          "style": Object {
            "borderRadius": "10%",
            "height": "200px",
            "padding": "5%",
            "width": "18rem",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h5
                  className="card-title white-text"
                />,
                <h6
                  className="card-subtitle mb-2 text-muted white-text"
                >
                  @ undefined
                </h6>,
                <p
                  className="card-text white-text"
                />,
                <p
                  className="card-text white-text"
                >
                  Estimated guest No.: 
                </p>,
              ],
              "className": "card-body",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "card-title white-text",
                },
                "ref": null,
                "rendered": null,
                "type": "h5",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "@ undefined",
                  "className": "card-subtitle mb-2 text-muted white-text",
                },
                "ref": null,
                "rendered": "@ undefined",
                "type": "h6",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "card-text white-text",
                },
                "ref": null,
                "rendered": null,
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Estimated guest No.: ",
                    undefined,
                  ],
                  "className": "card-text white-text",
                },
                "ref": null,
                "rendered": Array [
                  "Estimated guest No.: ",
                  undefined,
                ],
                "type": "p",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Cancel",
              "className": "btn btn-danger btn-sm btn-rounded",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Cancel",
            "type": "button",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Col
          m={4}
          s={10}
        >
          <div
            className="card elegant-color white-text"
            style={
              Object {
                "borderRadius": "10%",
                "height": "200px",
                "padding": "5%",
                "width": "18rem",
              }
            }
          >
            <div
              className="card-body"
            >
              <h5
                className="card-title white-text"
              />
              <h6
                className="card-subtitle mb-2 text-muted white-text"
              >
                @ undefined
              </h6>
              <p
                className="card-text white-text"
              />
              <p
                className="card-text white-text"
              >
                Estimated guest No.: 
              </p>
            </div>
            <button
              className="btn btn-danger btn-sm btn-rounded"
              onClick={[Function]}
            >
              Cancel
            </button>
          </div>
        </Col>,
        "style": Object {
          "display": "inlineBlock",
          "float": "left",
          "margin": "1%",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <div
            className="card elegant-color white-text"
            style={
              Object {
                "borderRadius": "10%",
                "height": "200px",
                "padding": "5%",
                "width": "18rem",
              }
            }
          >
            <div
              className="card-body"
            >
              <h5
                className="card-title white-text"
              />
              <h6
                className="card-subtitle mb-2 text-muted white-text"
              >
                @ undefined
              </h6>
              <p
                className="card-text white-text"
              />
              <p
                className="card-text white-text"
              >
                Estimated guest No.: 
              </p>
            </div>
            <button
              className="btn btn-danger btn-sm btn-rounded"
              onClick={[Function]}
            >
              Cancel
            </button>
          </div>,
          "m": 4,
          "s": 10,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="card-body"
              >
                <h5
                  className="card-title white-text"
                />
                <h6
                  className="card-subtitle mb-2 text-muted white-text"
                >
                  @ undefined
                </h6>
                <p
                  className="card-text white-text"
                />
                <p
                  className="card-text white-text"
                >
                  Estimated guest No.: 
                </p>
              </div>,
              <button
                className="btn btn-danger btn-sm btn-rounded"
                onClick={[Function]}
              >
                Cancel
              </button>,
            ],
            "className": "card elegant-color white-text",
            "style": Object {
              "borderRadius": "10%",
              "height": "200px",
              "padding": "5%",
              "width": "18rem",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h5
                    className="card-title white-text"
                  />,
                  <h6
                    className="card-subtitle mb-2 text-muted white-text"
                  >
                    @ undefined
                  </h6>,
                  <p
                    className="card-text white-text"
                  />,
                  <p
                    className="card-text white-text"
                  >
                    Estimated guest No.: 
                  </p>,
                ],
                "className": "card-body",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                    "className": "card-title white-text",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "h5",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "@ undefined",
                    "className": "card-subtitle mb-2 text-muted white-text",
                  },
                  "ref": null,
                  "rendered": "@ undefined",
                  "type": "h6",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                    "className": "card-text white-text",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Estimated guest No.: ",
                      undefined,
                    ],
                    "className": "card-text white-text",
                  },
                  "ref": null,
                  "rendered": Array [
                    "Estimated guest No.: ",
                    undefined,
                  ],
                  "type": "p",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Cancel",
                "className": "btn btn-danger btn-sm btn-rounded",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Cancel",
              "type": "button",
            },
          ],
          "type": "div",
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
